core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -138,9 +138,11 @@ class ControllerChannelManager(controllerContext: ControllerContext, config: Kaf
 
   private def removeExistingBroker(brokerState: ControllerBrokerStateInfo) {
     try {
+      // shutdown the RequestSendThread before closing the NetworkClient to
+      // avoid the concurrent use of the non-threadsafe classes as described in KAFKA-4959
+      brokerState.requestSendThread.shutdown()
```

Comment:
Alright I updated the PR with comments clarifying the thread-safety associated with the change.